### PR TITLE
Fix for https://github.com/huggingface/diffusers/issues/3836

### PR DESCRIPTION
**Pull request by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw**
# What does this PR do?

Fixes https://github.com/huggingface/diffusers/issues/3836 (issue)

## Before submitting
- [nope ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [x ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [x ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [nope ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
if this dont get merged until the base branch has conflicts.
i probably will not re-open a pr for this. instead will just mantain my own forked version with this fix and sync with master once in a while,so if posiible check and pull this pr soon.

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -1802,6 +1802,10 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
             tokenizer ([`~transformers.CLIPTokenizer`], *optional*, defaults to `None`):
                 An instance of `CLIPTokenizer` to use. If this parameter is `None`, the function loads a new instance
                 of `CLIPTokenizer` by itself if needed.
+	        mislljson_dir (`str`, *optional*, defaults to 'DIFFUSERS_CACHE'):
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:1805]: `tryharderfindmodel` ?

**Review by patrickvonplaten — COMMENTED** [review]
I don't understand this PR

File: src/diffusers/loaders.py (hunk 1 above)
Comment (cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw) [reply src/diffusers/loaders.py:1805]: i think that part is removed?but yeah the core is to try find the vae and etc.

**Comment by cm5kZGV2MjAyM3B1YmdpdGh1YmFjYw** [comment]
I dont think there is anything complex to understand.
Its just a patch for as the issue referred.
the lack of using cache_dir in all these recursive inner functions which some developer forget to pass it to all relevant functions.

eg,from_pretrained()

**Pull request closed without merging**